tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -185,6 +185,8 @@ in case you want to take a look under the hood.
 +------------+---------------------------------------------------------------------------------------------------------------------------------------------------+
 | as         | Cast the value to a given type if possible.                                                                                                       |
 +------------+---------------------------------------------------------------------------------------------------------------------------------------------------+
+| when       | Used by :ref:`pattern guard expressions<doc_gdscript_basics_pattern_guards>` in match_ statements.                                                |
```

Comment:
The table actually is currently sorted by pretty intuitive criteria:

1. **Control flow with parameters**: if, elif, else, for, while, match.
2. **Control flow statements**: continue, pass, return. 
3. **Class definition**: class, class_name, extends.
4. **Prepositions which are not control flow, or are *used in* control flow**: is, in, as.
5. Self, super.
6. **Definitions**: signal, func, static, const, enum, var.
7. **Misc**: breakpoint, preload, await, yield, assert, void.
8. **Constants**: PI, TAU, INF, NAN.

This order matches *roughly* the order you'd want to read things when introduced to the language.

I think that section 4 is right for "when". It's an English preposition but not used as a control flow statement itself. Compare its description with "in" in the table:
> Tests whether a value is within a string, array, range, dictionary, or node. When used with `for`, it iterates through them instead of testing.

I'm not worried about separating "when" from "match" in the table since you should pick up the term when you read the full "match" section of the page.

Reply:
Yes, this is one of the possible rationalizations. On the other hand, keywords can be divided into several subgroups: declarations, control flow, operators, other. And control flow keywords are clearly separated in the [source code](https://github.com/godotengine/godot/blob/e53dc80f2f9e0f5dd03103fe968662f7ffd258ba/modules/gdscript/gdscript.cpp#L2718-L2731) and highlighted in a separate color in the editor. `when` is a control flow keyword.

![](https://github.com/user-attachments/assets/ed9bcd36-c3bd-4d63-869b-9720dfc412db)

I do not insist on this order, I'm just explaining my logic. Perhaps in future PR we should split the keywords into subgroups and add headings to the table.

Reply:
Okay, I'm convinced by "when" being a flow control statement as far as the source is concerned. I moved it to right below "match" in the table, and changed it so "match" no longer links, like the nearby references to ``for``.